STACpopulator/implementations/CMIP6_UofT/add_CMIP6.py

```diff
@@ -88,14 +84,21 @@ def make_parser() -> argparse.ArgumentParser:
     parser.add_argument("stac_host", type=str, help="STAC API address")
     parser.add_argument("href", type=str, help="URL to a THREDDS catalog or a NCML XML with CMIP6 metadata.")
     parser.add_argument("--update", action="store_true", help="Update collection and its items")
-    parser.add_argument("--mode", choices=["full", "single"], default="full",
-                        help="Operation mode, processing the full dataset or only the single reference.")
     parser.add_argument(
-        "--config", type=str, help=(
+        "--mode",
+        choices=["full", "single"],
+        default="full",
+        help="Operation mode, processing the full dataset or only the single reference.",
+    )
+    parser.add_argument(
+        "--config",
+        type=str,
+        help=(
             "Override configuration file for the populator. "
             "By default, uses the adjacent configuration to the implementation class."
-        )
+        ),
     )
+    parser.add_argument("--debug", action="store_true", help="Set logger level to debug")
```

Comment:
I think it should be in the parent argparser (generic one before specific implementation is called). 
`ns.debug` will not exist if using other implementations than CMIP6

Reply:
Okay I've moved it. I'm not sure I like the current position much. Right now, one has to do:
```
stac-populator --debug run CMIP6_UofT stac-host catalog-url --auth-handler cookie --auth-identity ~/daccs_cookie.txt
```

I'd prefer if one could instead just do:

```
stac-populator run CMIP6_UofT stac-host catalog-url --auth-handler cookie --auth-identity ~/daccs_cookie.txt --debug
```

By the way, this might be related to #46. I noticed that there is something off with the CLI parser system you've put into place. I'll let you look into it since you put it in and hence you probably understand it better than me.

Reply:
To allow `stac-populator run CMIP6_UofT --debug ...`, you could define the logging options as their own `log_parser = argparse.ArgumentParser(...)` and pass that `log_parser` in `parents` (https://github.com/crim-ca/stac-populator/blob/24eeb6b0b863dd5a19f1903527300903b129f9ca/STACpopulator/cli.py#L163). Need to pass it down also to `parser_maker` (https://github.com/crim-ca/stac-populator/blob/24eeb6b0b863dd5a19f1903527300903b129f9ca/STACpopulator/cli.py#L158C32-L158C44) so it can add `log_parser` to its own `parents` as well.